bundles/org.openhab.binding.weathercompany/src/main/java/org/openhab/binding/weathercompany/internal/handler/WeatherCompanyAbstractHandler.java

```diff
@@ -0,0 +1,399 @@
+                    break;
+                case HttpStatus.NOT_FOUND_404:
+                    errorMsg = "HTTP response 404: Endpoint not found";
+                    break;
+                case HttpStatus.METHOD_NOT_ALLOWED_405:
+                    errorMsg = "HTTP response 405: Method not allowed";
+                    break;
+                case HttpStatus.NOT_ACCEPTABLE_406:
+                    errorMsg = "HTTP response 406: Not acceptable";
+                    break;
+                case HttpStatus.REQUEST_TIMEOUT_408:
+                    errorMsg = "HTTP response 408: Request timeout";
+                    break;
+                case HttpStatus.INTERNAL_SERVER_ERROR_500:
+                    errorMsg = "HTTP response 500: Internal server error";
+                    break;
+                case HttpStatus.BAD_GATEWAY_502:
+                case HttpStatus.SERVICE_UNAVAILABLE_503:
+                case HttpStatus.GATEWAY_TIMEOUT_504:
+                    errorMsg = String.format("HTTP response %d: Service unavailable or gateway issue",
+                            contentResponse.getStatus());
+                    break;
+                default:
+                    errorMsg = String.format("HTTP GET failed: %d, %s", contentResponse.getStatus(),
+                            contentResponse.getReason());
+                    break;
+            }
+        } catch (TimeoutException e) {
+            errorMsg = "TimeoutException: Call to Weather Company API timed out";
+        } catch (ExecutionException e) {
+            errorMsg = String.format("ExecutionException: %s", e.getMessage());
+        } catch (InterruptedException e) {
+            errorMsg = String.format("InterruptedException: %s", e.getMessage());
+        }
+        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, errorMsg);
+        logger.debug("{}", errorMsg);
+        return null;
+    }
+
+    /*
+     * Convert UTC Unix epoch seconds to local time
+     */
+    protected DateTimeType getLocalDateTimeType(long epochSeconds) {
+        Instant instant = Instant.ofEpochSecond(epochSeconds);
+        ZonedDateTime localDateTime = instant.atZone(getZoneId());
+        DateTimeType dateTimeType = new DateTimeType(localDateTime);
+        return dateTimeType;
+    }
+
+    /*
+     * Convert UTC time string to local time
+     * Input string is of form 2018-12-02T10:47:00.000Z
+     */
+    protected DateTimeType getLocalDateTimeType(String dateTimeString) {
+        DateTimeType dateTimeType;
+        try {
+            Instant instant = Instant.parse(dateTimeString);
+            ZonedDateTime localDateTime = instant.atZone(getZoneId());
+            dateTimeType = new DateTimeType(localDateTime);
+        } catch (DateTimeParseException e) {
+            logger.debug("Error parsing date/time string: {}", e.getMessage());
+            dateTimeType = new DateTimeType();
+        } catch (IllegalArgumentException e) {
```

Comment:
In what case is this specific exception thrown? It sounds like a case which should be prevented IMHO

Reply:
IIRC, I originally was using the DateTimeType(String) constructor, which threw the IllegalArgumentException [here](https://github.com/openhab/openhab-core/blob/10acf5cf4675ba908ba69361df951bf36e9744a6/bundles/org.openhab.core.compat1x/src/main/java/org/openhab/core/library/types/DateTimeType.java#L54). However, that was not giving me the correct local time based on the openHAB timezone. I changed the implementation, but forgot to remove this catch. I think it can safely be removed.